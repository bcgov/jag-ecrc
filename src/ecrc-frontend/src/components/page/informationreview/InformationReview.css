@media (max-width: 575.98px) {
  .info-buttons {
    flex-direction: column;
  }

  .btn {
    margin-right: 0px;
    margin-bottom: 5px;
  }
}
