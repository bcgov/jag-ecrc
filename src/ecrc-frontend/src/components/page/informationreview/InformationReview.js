import React, { useState, useEffect } from "react";
import axios from "axios";
import { Redirect } from "react-router-dom";
import PropTypes from "prop-types";

import Header from "../../base/header/Header";
import Footer from "../../base/footer/Footer";
import Table from "../../composite/table/Table";
import { Button } from "../../base/button/Button";
import SideCards from "../../composite/sideCards/SideCards";
import {
  generateJWTToken,
  accessJWTToken,
  isActionPerformed,
  isAuthorized
} from "../../../modules/AuthenticationHelper";

export default function InformationReview({
  page: {
    header,
    applicant: {
      legalFirstNm,
      legalSecondNm,
      legalSurnameNm,
      alias1FirstNm,
      alias1SecondNm,
      alias1SurnameNm,
      alias2FirstNm,
      alias2SecondNm,
      alias2SurnameNm,
      alias3FirstNm,
      alias3SecondNm,
      alias3SurnameNm,
      birthDt,
      genderTxt,
      addressLine1,
      cityNm,
      provinceNm,
      postalCodeTxt,
      countryNm,
      mailingAddressLine1,
      mailingCity,
      mailingProvince,
      mailingPostalCode,
      birthPlace,
      driversLicNo,
      phoneNumber,
      emailAddress,
      jobTitle,
      organizationFacility
    },
    org: {
      orgApplicantRelationship,
      orgTicketNumber,
      defaultScheduleTypeCd,
      defaultCrcScopeLevelCd
    },
    setApplicationInfo,
    saveApplicant,
    saveOrg,
    saveApplicationInfo,
    setError
  }
}) {
  const [toBack, setToBack] = useState(false);
  const [toHome, setToHome] = useState(false);
  const [toSuccess, setToSuccess] = useState(false);
  const [toError, setToError] = useState(false);
  const [boxChecked, setBoxChecked] = useState(false);

  useEffect(() => {
    window.scrollTo(0, 0);

    if (!isAuthorized() || !isActionPerformed("appForm")) {
      setToHome(true);
      console.log("Redirected by routing");
    }
  }, []);

  const personalInfoElement = [
    { name: "First Name", value: legalFirstNm },
    {
      name: "Middle Name",
      value: legalSecondNm
    },
    {
      name: "Last Name",
      value: legalSurnameNm
    },
    {
      name: "City and Country of Birth",
      value: birthPlace
    },
    {
      name: "Birth Date",
      value: birthDt
    },
    {
      name: "Sex",
      value: genderTxt
    }
  ];

  if (driversLicNo) {
    personalInfoElement.push({
      name: "BC Driver's Licence",
      value: driversLicNo
    });
  }

  const personalInfoTable = {
    header: "PERSONAL INFORMATION",
    tableElements: personalInfoElement
  };

  const previousNamesElement = [];

  if (alias1FirstNm || alias1SecondNm || alias1SurnameNm) {
    previousNamesElement.push({
      key: "alias1FirstNm",
      name: "First Name",
      value: alias1FirstNm
    });
    previousNamesElement.push({
      key: "alias1SecondNm",
      name: "Middle Name",
      value: alias1SecondNm
    });
    previousNamesElement.push({
      key: "alias1SurnameNm",
      name: "Last Name",
      value: alias1SurnameNm
    });
  }

  if (alias2FirstNm || alias2SecondNm || alias2SurnameNm) {
    previousNamesElement.push({
      key: "alias2FirstNm",
      name: "First Name",
      value: alias2FirstNm
    });
    previousNamesElement.push({
      key: "alias2SecondNm",
      name: "Middle Name",
      value: alias2SecondNm
    });
    previousNamesElement.push({
      key: "alias2SurnameNm",
      name: "Last Name",
      value: alias2SurnameNm
    });
  }

  if (alias3FirstNm || alias3SecondNm || alias3SurnameNm) {
    previousNamesElement.push({
      key: "alias3FirstNm",
      name: "First Name",
      value: alias3FirstNm
    });
    previousNamesElement.push({
      key: "alias3SecondNm",
      name: "Middle Name",
      value: alias3SecondNm
    });
    previousNamesElement.push({
      key: "alias3SurnameNm",
      name: "Last Name",
      value: alias3SurnameNm
    });
  }

  const previousNamesTable = {
    header: (
      <div>
        PREVIOUS NAMES
        <span className="smallTableHeader">
          (Including birth name, previous name, maiden name, and alias)
        </span>
      </div>
    ),
    tableElements: previousNamesElement
  };

  const positionInfoElement = [
    {
      name: "Your Position/Job Title",
      value: jobTitle
    }
  ];

  if (organizationFacility) {
    positionInfoElement.push({
      name: "Organization Facility",
      value: organizationFacility
    });
  }

  const positionInfoTable = {
    header: "POSITION WITH ORGANIZATION",
    tableElements: positionInfoElement
  };

  const contactElement = [
    {
      name: "Residential Address",
      value: addressLine1
    },
    {
      name: "City",
      value: cityNm
    },
    {
      name: "Province",
      value: provinceNm
    },
    {
      name: "Postal Code",
      value: postalCodeTxt
    },
    {
      name: "Country",
      value: countryNm
    },
    {
      name: "Primary Phone Number",
      value: phoneNumber
    },
    {
      name: "Personal Email Address",
      value: emailAddress
    }
  ];

  const contactTable = {
    header: "CONTACT INFORMATION",
    tableElements: contactElement
  };

  const mailingAddressElement = [
    {
      name: "Mailing Address",
      value: mailingAddressLine1
    },
    {
      name: "City",
      value: mailingCity
    },
    {
      name: "Province",
      value: mailingProvince
    },
    {
      name: "Postal Code",
      value: mailingPostalCode
    },
    {
      name: "Country",
      value: countryNm
    }
  ];

  const mailingAddressTable = {
    header: "MAILING ADDRESS",
    tableElements: mailingAddressElement
  };

  const confirmButton = {
    label: "SUBMIT",
    buttonStyle: "btn ecrc_go_btn",
    buttonSize: "btn btn-sm",
    type: "submit",
    disabled: !boxChecked
  };

  const confirm = () => {
    // TODO: Check if volunteer, if yes, success, else, cont.
    // CALL THAT API
    const token = sessionStorage.getItem("jwt");
    const uuid = sessionStorage.getItem("uuid");

    const createApplicantInfo = {
      orgTicketNumber,
      requestGuid: uuid,
      callPurpose: "CRC",
      legalSurnameNm,
      legalFirstNm,
      legalSecondNm,
      birthDt,
      genderTxt,
      birthPlace,
      alias1FirstNm,
      alias1SecondNm,
      alias1SurnameNm,
      alias2FirstNm,
      alias2SecondNm,
      alias2SurnameNm,
      alias3FirstNm,
      alias3SecondNm,
      alias3SurnameNm,
      phoneNumber,
      addressLine1: mailingAddressLine1,
      cityNm: mailingCity,
      provinceNm: mailingProvince,
      countryNm,
      postalCodeTxt: mailingPostalCode,
      driversLicNo
    };

    let partyId;
    let sessionId;
    let invoiceId;
    let serviceFeeAmount;
    let serviceId;

    Promise.all([
      axios.post("/ecrc/private/createApplicant", createApplicantInfo, {
        headers: {
          Authorization: `Bearer ${token}`
        }
      }),
      // .catch(error => {
      //   setToError(true);
      //   setError(error.response.status.toString());
      // }),
      axios
        .get(
          `/ecrc/private/getNextSessionId?orgTicketNumber=${orgTicketNumber}&requestGuid=${uuid}`,
          {
            headers: {
              Authorization: `Bearer ${token}`
            }
          }
        )
        .catch(error => {
          setToError(true);
          setError(error.response.status.toString());
        }),
      axios
        .get(
          `/ecrc/private/getNextInvoiceId?orgTicketNumber=${orgTicketNumber}&requestGuid=${uuid}`,
          {
            headers: {
              Authorization: `Bearer ${token}`
            }
          }
        )
        .catch(error => {
          setToError(true);
          setError(error.response.status.toString());
        }),
      axios
        .get(
          `/ecrc/private/getServiceFeeAmount?orgTicketNumber=${orgTicketNumber}&scheduleTypeCd=${defaultScheduleTypeCd}&scopeLevelCd=${defaultCrcScopeLevelCd}&requestGuid=${uuid}`,
          {
            headers: {
              Authorization: `Bearer ${token}`
            }
          }
        )
        .catch(error => {
          setToError(true);
          setError(error.response.status.toString());
        })
    ])
      .then(all => {
        partyId = all[0].data.partyId;
        sessionId = all[1].data.sessionId;
        invoiceId = all[2].data.invoiceId;
        serviceFeeAmount = all[3].data.serviceFeeAmount;

        // NEED CLARIFICATION: - as per Jason Lee, awaiting confirmation
        // eivPassDetailsResults - String returned from equifax, see Shaun
        const newCRC = {
          orgTicketNumber,
          requestGuid: uuid,
          schedule_Type_Cd: defaultScheduleTypeCd,
          scope_Level_Cd: defaultCrcScopeLevelCd,
          appl_Party_Id: partyId,
          org_Appl_To_Pay: "A",
          applicant_Posn: jobTitle,
          child_Care_Fac_Nm: "child_Care_Fac_Nm",
          governing_Body_Nm: "governing_Body_Nm",
          session_Id: sessionId,
          invoice_Id: invoiceId,
          auth_Release_EIV_Vendor_YN: "Y",
          auth_Conduct_CRC_Check_YN: "Y",
          auth_Release_To_Org_YN: "Y",
          appl_Identity_Verified_EIV_YN: "Y",
          eivPassDetailsResults: "eivPassDetailsResults"
        };

        return axios.post("/ecrc/private/createNewCRCService", newCRC, {
          headers: {
            Authorization: `Bearer ${token}`
          }
        });
      })
      .then(crcResponse => {
        serviceId = crcResponse.data.serviceId;

        const appInfo = {
          partyId,
          sessionId,
          invoiceId,
          serviceFeeAmount,
          serviceId
        };

        setApplicationInfo(appInfo);

        if (orgApplicantRelationship === "VOLUNTEER") {
          setToSuccess(true);
        } else {
          saveApplicant();
          saveOrg();
          saveApplicationInfo(appInfo);

          const createURL = {
            invoiceNumber: invoiceId,
            requestGuid: uuid,
            approvedPage: `${process.env.REACT_APP_FRONTEND_BASE_URL}/ecrc/success`,
            declinedPage: `${process.env.REACT_APP_FRONTEND_BASE_URL}/ecrc/success`,
            errorPage: `${process.env.REACT_APP_FRONTEND_BASE_URL}/ecrc/success`,
            totalItemsAmount: serviceFeeAmount,
            serviceIdRef1: serviceId,
            partyIdRef2: partyId
          };

          const currentPayload = accessJWTToken(sessionStorage.getItem("jwt"));
          const actionsPerformed = [
            ...currentPayload.actionsPerformed,
            "infoReview"
          ];
          const newPayload = {
            ...currentPayload,
            actionsPerformed
          };
          generateJWTToken(newPayload);

          axios
            .post("/ecrc/private/createPaymentUrl", createURL, {
              headers: {
                Authorization: `Bearer ${token}`
              }
            })
            .then(urlResponse => {
              window.location.href = urlResponse.data.paymentUrl;
            })
            .catch(error => {
              setToError(true);
              setError(error.response.status.toString());
            });
        }
      })
      .catch(error => {
        setToError(true);
        setError(error.response.status.toString());
      });
  };

  const cancelButton = {
    label: "EDIT APPLICATION",
    buttonStyle: "btn ecrc_accessary_btn",
    buttonSize: "btn btn-sm",
    type: "submit"
  };

  const edit = () => {
    setToBack(true);
  };

  if (toBack) {
    return <Redirect to="/criminalrecordcheck/applicationform" />;
  }

  if (toSuccess) {
    const currentPayload = accessJWTToken(sessionStorage.getItem("jwt"));
    const actionsPerformed = [...currentPayload.actionsPerformed, "infoReview"];
    const newPayload = {
      ...currentPayload,
      actionsPerformed
    };
    generateJWTToken(newPayload);
    return <Redirect to="/criminalrecordcheck/success" />;
  }

  if (toHome) {
    return <Redirect to="/" />;
  }

  if (toError) {
    return <Redirect to="/criminalrecordcheck/success" />;
  }

  return (
    <main>
      <Header header={header} />
      <div className="page">
        <div className="content col-md-8">
          <h1>Information Review</h1>
          <p>
            Please confirm that the information provided below is accurate. If
            it is not, please select Edit Application.
          </p>
          <Table table={personalInfoTable} />
          {previousNamesElement.length > 0 ? (
            <Table table={previousNamesTable} />
          ) : null}
          <Table table={positionInfoTable} />
          <Table table={contactTable} />
          <Table table={mailingAddressTable} />
          <div className="declareTitle">DECLARATION</div>
          <section className="declareSection">
            <input
              type="checkbox"
              onClick={() => {
                setBoxChecked(!boxChecked);
              }}
            />
            <span className="declaration-cb">
              I certify that, to the best of my knowledge, the information I
              have provided on my application is complete, honest and accurate.
              Any false statements or deliberate omissions on a consent form
              filed with the CRRP may result in the inability of the CRRP to
              accurately determine whether the applicant poses a risk to
              children or vulnerable adults.
            </span>
          </section>
          <div className="buttons">
            <Button button={cancelButton} onClick={edit} />
            <Button button={confirmButton} onClick={confirm} />
          </div>
        </div>
        <div className="sidecard">
          <SideCards type={"personalinformation"} />
        </div>
      </div>
      <Footer />
    </main>
  );
}

InformationReview.propTypes = {
  page: PropTypes.shape({
    header: PropTypes.shape({
      name: PropTypes.string.isRequired
    }),
    applicant: PropTypes.shape({
      legalFirstNm: PropTypes.string.isRequired,
      legalSecondNm: PropTypes.string.isRequired,
      legalSurnameNm: PropTypes.string.isRequired,
      alias1FirstNm: PropTypes.string,
      alias1SecondNm: PropTypes.string,
      alias1SurnameNm: PropTypes.string,
      alias2FirstNm: PropTypes.string,
      alias2SecondNm: PropTypes.string,
      alias2SurnameNm: PropTypes.string,
      alias3FirstNm: PropTypes.string,
      alias3SecondNm: PropTypes.string,
      alias3SurnameNm: PropTypes.string,
      birthDt: PropTypes.string.isRequired,
      genderTxt: PropTypes.string.isRequired,
      addressLine1: PropTypes.string.isRequired,
      cityNm: PropTypes.string.isRequired,
      provinceNm: PropTypes.string.isRequired,
      postalCodeTxt: PropTypes.string.isRequired,
      countryNm: PropTypes.string.isRequired,
      mailingAddressLine1: PropTypes.string.isRequired,
      mailingCity: PropTypes.string.isRequired,
      mailingProvince: PropTypes.string.isRequired,
      mailingPostalCode: PropTypes.string.isRequired,
      birthPlace: PropTypes.string.isRequired,
      driversLicNo: PropTypes.string,
      phoneNumber: PropTypes.string.isRequired,
      emailAddress: PropTypes.string.isRequired,
      jobTitle: PropTypes.string.isRequired,
      organizationFacility: PropTypes.string
    }),
    org: PropTypes.shape({
      orgApplicantRelationship: PropTypes.string.isRequired,
      orgTicketNumber: PropTypes.string.isRequired,
      defaultScheduleTypeCd: PropTypes.string.isRequired,
      defaultCrcScopeLevelCd: PropTypes.string.isRequired
    }),
    setApplicationInfo: PropTypes.func.isRequired,
    saveApplicant: PropTypes.func.isRequired,
    saveOrg: PropTypes.func.isRequired,
    saveApplicationInfo: PropTypes.func.isRequired
  })
};

InformationReview.defaultProps = {
  page: {
    applicant: {
      alias1FirstNm: "",
      alias1SecondNm: "",
      alias1SurnameNm: "",
      alias2FirstNm: "",
      alias2SecondNm: "",
      alias2SurnameNm: "",
      alias3FirstNm: "",
      alias3SecondNm: "",
      alias3SurnameNm: ""
    }
  }
};
