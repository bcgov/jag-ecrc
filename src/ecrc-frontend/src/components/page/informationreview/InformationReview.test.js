import React from "react";
import { create } from "react-test-renderer";
import { MemoryRouter, Router } from "react-router-dom";
import {
  render,
  fireEvent,
  getByRole,
  getByText,
  wait
} from "@testing-library/react";
import { createMemoryHistory } from "history";
import axios from "axios";

import InformationReview from "./InformationReview";
import { generateJWTToken } from "../../../modules/AuthenticationHelper";

// Mock axios
jest.mock("axios");

describe("InformationReview Component", () => {
  const header = {
    name: "Criminal Record Check"
  };

  const applicant = {
    legalFirstNm: "Robert",
    legalSecondNm: "Norman",
    legalSurnameNm: "Ross",
    birthPlace: "Daytona Beach, Florida",
    alias1FirstNm: "Robert",
    alias1SecondNm: "Norman",
    alias1SurnameNm: "Ross",
    alias2FirstNm: "Robert",
    alias2SecondNm: "Norman",
    alias2SurnameNm: "Ross",
    alias3FirstNm: "Robert",
    alias3SecondNm: "Norman",
    alias3SurnameNm: "Ross",
    birthDt: "1942-10-29",
    genderTxt: "Male",
    driversLicNo: "1234567",
    phoneNumber: "2501234567",
    emailAddress: "bob.ross@example.com",
    addressLine1: "123 Somewhere",
    cityNm: "Here",
    provinceNm: "British Columbia",
    postalCodeTxt: "V9V 9V9",
    countryNm: "Canada",
    mailingAddressLine1: "456 Elsewhere",
    mailingCity: "There",
    mailingProvince: "Ontario",
    mailingPostalCode: "V1V 1A1",
    jobTitle: "Painter",
    organizationFacility: "Something"
  };

  const org = {
    orgApplicantRelationship: "EMPLOYEE",
    orgTicketNumber: "crce",
    defaultScheduleTypeCd: "WBSD",
    defaultCrcScopeLevelCd: "WWCH"
  };

  const setApplicationInfo = jest.fn();
  const saveApplicant = jest.fn();
  const saveOrg = jest.fn();
  const saveApplicationInfo = jest.fn();
  const setError = jest.fn();

  window.scrollTo = jest.fn();

  // Mock window location
  const mockWindow = jest.fn();
  delete window.location;
  window.location = { assign: mockWindow };

  beforeEach(() => {
    sessionStorage.setItem("validator", "secret");
    sessionStorage.setItem("uuid", "unique123");
    generateJWTToken({
      actionsPerformed: ["appForm"],
      authorities: ["Authorized"]
    });
  });

  test("Matches the snapshot", () => {
    const page = {
      header,
      applicant,
      org,
      setApplicationInfo,
      saveApplicant,
      saveOrg,
      saveApplicationInfo,
      setError
    };
    const infoReview = create(
      <MemoryRouter>
        <InformationReview page={page} />
      </MemoryRouter>
    );
    expect(infoReview.toJSON()).toMatchSnapshot();
  });

  test("Validate checkbox", () => {
    const page = {
      header,
      applicant,
      org,
      setApplicationInfo,
      saveApplicant,
      saveOrg,
      saveApplicationInfo,
      setError
    };
    const history = createMemoryHistory();
    const { container } = render(
      <Router history={history}>
        <InformationReview page={page} />
      </Router>
    );

    expect(getByText(container, "SUBMIT").disabled).toBeTruthy();

    fireEvent.click(getByRole(container, "checkbox"));

    expect(getByText(container, "SUBMIT").disabled).toBeFalsy();
  });

  test("Validate Back button", () => {
    const page = {
      header,
      applicant,
      org,
      setApplicationInfo,
      saveApplicant,
      saveOrg,
      saveApplicationInfo,
      setError
    };
    const history = createMemoryHistory();
    const { container } = render(
      <Router history={history}>
        <InformationReview page={page} />
      </Router>
    );
    fireEvent.click(getByText(container, "EDIT APPLICATION"));
    expect(history.location.pathname).toEqual(
      "/criminalrecordcheck/applicationform"
    );
  });

  test("Validate Employee relationship flow", async () => {
    const page = {
      header,
      applicant,
      org,
      setApplicationInfo,
      saveApplicant,
      saveOrg,
      saveApplicationInfo,
      setError
    };
    axios.get.mockImplementation(() =>
      Promise.resolve({
        data: {
          sessionId: "123",
          invoiceId: "123",
          serviceFeeAmount: "123"
        }
      })
    );

    axios.post.mockImplementation(() =>
      Promise.resolve({
        data: {
          partyId: "123",
          serviceId: "123",
          urlResponse: "http://sample.com"
        }
      })
    );

    const history = createMemoryHistory();
    const { container } = render(
      <Router history={history}>
        <InformationReview page={page} />
      </Router>
    );
    fireEvent.click(getByRole(container, "checkbox"));
    fireEvent.click(getByText(container, "SUBMIT"));

    await wait(() => {
      expect(setApplicationInfo).toHaveBeenCalled();
      expect(saveApplicant).toHaveBeenCalled();
      expect(saveOrg).toHaveBeenCalled();
      expect(saveApplicationInfo).toHaveBeenCalled();
    });
  });

  test("Validate Volunteer relationship flow", async () => {
    applicant.driversLicNo = "";
    applicant.alias1FirstNm = "";
    applicant.alias1SecondNm = "";
    applicant.alias1SurnameNm = "";
    applicant.alias2FirstNm = "";
    applicant.alias2SecondNm = "";
    applicant.alias2SurnameNm = "";
    applicant.alias3FirstNm = "";
    applicant.alias3SecondNm = "";
    applicant.alias3SurnameNm = "";

    applicant.organizationFacility = "";

    org.orgApplicantRelationship = "VOLUNTEER";

    const page = {
      header,
      applicant,
      org,
      setApplicationInfo,
      saveApplicant,
      saveOrg,
      saveApplicationInfo,
      setError
    };

    axios.get.mockImplementation(() =>
      Promise.resolve({
        data: {
          sessionId: "123",
          invoiceId: "123",
          serviceFeeAmount: "123"
        }
      })
    );

    axios.post.mockImplementation(() =>
      Promise.resolve({
        data: {
          partyId: "123",
          serviceId: "123",
          urlResponse: "http://sample.com"
        }
      })
    );

    const history = createMemoryHistory();
    const { container } = render(
      <Router history={history}>
        <InformationReview page={page} />
      </Router>
    );
    fireEvent.click(getByRole(container, "checkbox"));
    fireEvent.click(getByText(container, "SUBMIT"));

    await wait(() => {
      expect(setApplicationInfo).toHaveBeenCalled();
      expect(history.location.pathname).toEqual("/criminalrecordcheck/success");
    });
  });
});
