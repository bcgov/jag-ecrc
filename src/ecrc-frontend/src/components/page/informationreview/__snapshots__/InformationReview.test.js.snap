// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationReview Component Matches the snapshot 1`] = `
<main>
  <header>
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-none d-md-block"
            height="44"
            src="/images/bcid-logo-rev-en.svg"
            width="181"
          />
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-md-none"
            height="44"
            src="/images/bcid-symbol-rev.svg"
            width="64"
          />
        </a>
        <div>
          <a
            className="navbar-brand nav-item nav-link"
            href="/"
          >
            Criminal Record Check
          </a>
        </div>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarNavAltMarkup"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        />
      </div>
    </nav>
  </header>
  <div
    className="page"
  >
    <div
      className="content col-md-8"
    >
      <h1>
        Information Review
      </h1>
      <p>
        Please confirm that the information provided below is accurate. If it is not, please select Edit Application.
      </p>
      <table>
        <thead>
          <tr>
            <th
              colSpan="2"
            >
              PERSONAL INFORMATION
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              First Name
            </td>
            <td>
              Robert
            </td>
          </tr>
          <tr>
            <td>
              Middle Name
            </td>
            <td>
              Norman
            </td>
          </tr>
          <tr>
            <td>
              Last Name
            </td>
            <td>
              Ross
            </td>
          </tr>
          <tr>
            <td>
              City and Country of Birth
            </td>
            <td>
              Daytona Beach, Florida
            </td>
          </tr>
          <tr>
            <td>
              Birth Date
            </td>
            <td>
              1942-10-29
            </td>
          </tr>
          <tr>
            <td>
              Sex
            </td>
            <td>
              Male
            </td>
          </tr>
          <tr>
            <td>
              BC Driver's Licence
            </td>
            <td>
              1234567
            </td>
          </tr>
        </tbody>
      </table>
      <table>
        <thead>
          <tr>
            <th
              colSpan="2"
            >
              <div>
                PREVIOUS NAMES
                <span
                  className="smallTableHeader"
                >
                  (Including birth name, previous name, maiden name, and alias)
                </span>
              </div>
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              First Name
            </td>
            <td>
              Robert
            </td>
          </tr>
          <tr>
            <td>
              Middle Name
            </td>
            <td>
              Norman
            </td>
          </tr>
          <tr>
            <td>
              Last Name
            </td>
            <td>
              Ross
            </td>
          </tr>
          <tr>
            <td>
              First Name
            </td>
            <td>
              Robert
            </td>
          </tr>
          <tr>
            <td>
              Middle Name
            </td>
            <td>
              Norman
            </td>
          </tr>
          <tr>
            <td>
              Last Name
            </td>
            <td>
              Ross
            </td>
          </tr>
          <tr>
            <td>
              First Name
            </td>
            <td>
              Robert
            </td>
          </tr>
          <tr>
            <td>
              Middle Name
            </td>
            <td>
              Norman
            </td>
          </tr>
          <tr>
            <td>
              Last Name
            </td>
            <td>
              Ross
            </td>
          </tr>
        </tbody>
      </table>
      <table>
        <thead>
          <tr>
            <th
              colSpan="2"
            >
              POSITION WITH ORGANIZATION
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Your Position/Job Title
            </td>
            <td>
              Painter
            </td>
          </tr>
          <tr>
            <td>
              Organization Facility
            </td>
            <td>
              Something
            </td>
          </tr>
        </tbody>
      </table>
      <table>
        <thead>
          <tr>
            <th
              colSpan="2"
            >
              CONTACT INFORMATION
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Residential Address
            </td>
            <td>
              123 Somewhere
            </td>
          </tr>
          <tr>
            <td>
              City
            </td>
            <td>
              Here
            </td>
          </tr>
          <tr>
            <td>
              Province
            </td>
            <td>
              British Columbia
            </td>
          </tr>
          <tr>
            <td>
              Postal Code
            </td>
            <td>
              V9V 9V9
            </td>
          </tr>
          <tr>
            <td>
              Country
            </td>
            <td>
              Canada
            </td>
          </tr>
          <tr>
            <td>
              Primary Phone Number
            </td>
            <td>
              2501234567
            </td>
          </tr>
          <tr>
            <td>
              Personal Email Address
            </td>
            <td>
              bob.ross@example.com
            </td>
          </tr>
        </tbody>
      </table>
      <table>
        <thead>
          <tr>
            <th
              colSpan="2"
            >
              MAILING ADDRESS
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Mailing Address
            </td>
            <td>
              456 Elsewhere
            </td>
          </tr>
          <tr>
            <td>
              City
            </td>
            <td>
              There
            </td>
          </tr>
          <tr>
            <td>
              Province
            </td>
            <td>
              Ontario
            </td>
          </tr>
          <tr>
            <td>
              Postal Code
            </td>
            <td>
              V1V 1A1
            </td>
          </tr>
          <tr>
            <td>
              Country
            </td>
            <td>
              Canada
            </td>
          </tr>
        </tbody>
      </table>
      <div
        className="declareTitle"
      >
        DECLARATION
      </div>
      <section
        className="declareSection"
      >
        <input
          onClick={[Function]}
          type="checkbox"
        />
        <span
          className="declaration-cb"
        >
          I certify that, to the best of my knowledge, the information I have provided on my application is complete, honest and accurate. Any false statements or deliberate omissions on a consent form filed with the CRRP may result in the inability of the CRRP to accurately determine whether the applicant poses a risk to children or vulnerable adults.
        </span>
      </section>
      <div
        className="buttons"
      >
        <button
          className="btn ecrc_accessary_btn btn btn-sm"
          onClick={[Function]}
          type="submit"
        >
          EDIT APPLICATION
          <div
            className="loader-hide"
          >
            <div
              className="btn-loader"
            />
          </div>
        </button>
        <button
          className="btn ecrc_go_btn btn btn-sm"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          SUBMIT
          <div
            className="loader-hide"
          >
            <div
              className="btn-loader"
            />
          </div>
        </button>
      </div>
    </div>
    <div
      className="sidecard"
    >
      <div>
        <div
          className="dashboard-spacing"
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="row"
          >
            <div
              className="col-12"
              style={
                Object {
                  "paddingTop": "30px",
                  "position": "relative",
                }
              }
            >
              <p />
              <section
                className="submit-container"
                id="blue-section"
              >
                <h2
                  className="heading-style"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="round-icon-wrapper"
                    >
                      <svg
                        className="side-card-icon"
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        strokeWidth="0"
                        style={
                          Object {
                            "color": undefined,
                          }
                        }
                        viewBox="0 0 384 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M202.021 0C122.202 0 70.503 32.703 29.914 91.026c-7.363 10.58-5.093 25.086 5.178 32.874l43.138 32.709c10.373 7.865 25.132 6.026 33.253-4.148 25.049-31.381 43.63-49.449 82.757-49.449 30.764 0 68.816 19.799 68.816 49.631 0 22.552-18.617 34.134-48.993 51.164-35.423 19.86-82.299 44.576-82.299 106.405V320c0 13.255 10.745 24 24 24h72.471c13.255 0 24-10.745 24-24v-5.773c0-42.86 125.268-44.645 125.268-160.627C377.504 66.256 286.902 0 202.021 0zM192 373.459c-38.196 0-69.271 31.075-69.271 69.271 0 38.195 31.075 69.27 69.271 69.27s69.271-31.075 69.271-69.271-31.075-69.27-69.271-69.27z"
                        />
                      </svg>
                    </div>
                    <div
                      className="sideCardTitle"
                    >
                      Update your BC Services Card Information
                    </div>
                  </div>
                </h2>
                <div
                  className="submit-content"
                >
                  <span>
                    If you find an error with your:
                    <ul
                      className="discStyle"
                    >
                      <li>
                        First Name
                      </li>
                      <li>
                        Middle Name
                      </li>
                      <li>
                        Last Name
                      </li>
                      <li>
                        Date of Birth
                      </li>
                      <li>
                        Sex
                      </li>
                      <li>
                        Current Street Address
                      </li>
                    </ul>
                    Please contact either 
                    <a
                      className="link"
                      href="https://www2.gov.bc.ca/gov/content/governments/organizational-structure/ministries-organizations/ministries/citizens-services/servicebc"
                    >
                      Service BC
                    </a>
                    , 
                    <a
                      className="link"
                      href="https://www.icbc.com/Pages/default.aspx"
                    >
                      ICBC
                    </a>
                     or 
                    <a
                      className="link"
                      href="https://www.addresschange.gov.bc.ca/"
                    >
                      AddressChangeBC
                    </a>
                     to correct it.
                  </span>
                </div>
              </section>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="footer"
  >
    <nav
      className="navbar navbar-expand-sm navbar-dark"
    >
      <div
        className="container"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
              rel="noopener noreferrer"
              target="_blank"
            >
              Disclaimer
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/privacy"
              rel="noopener noreferrer"
              target="_blank"
            >
              Privacy
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/accessibility"
              rel="noopener noreferrer"
              target="_blank"
            >
              Accessibility
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/copyright"
              rel="noopener noreferrer"
              target="_blank"
            >
              Copyright
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/contact-us"
              rel="noopener noreferrer"
              target="_blank"
            >
              Contact Us
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </footer>
</main>
`;
